homeassistant/components/feedreader/config_flow.py

```diff
@@ -0,0 +1,137 @@
+from typing import TYPE_CHECKING, Any
+import urllib.error
+
+import feedparser
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry, ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_URL
+from homeassistant.core import HomeAssistant
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.selector import (
+    TextSelector,
+    TextSelectorConfig,
+    TextSelectorType,
+)
+
+from .const import CONF_MAX_ENTRIES, DEFAULT_MAX_ENTRIES, DOMAIN
+
+LOGGER = logging.getLogger(__name__)
+
+
+async def async_fetch_feed(hass: HomeAssistant, url: str) -> feedparser.FeedParserDict:
+    """Fetch the feed."""
+    return await hass.async_add_executor_job(feedparser.parse, url)
+
+
+class FlowHandler(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow."""
+
+    VERSION = 1
+    _config_entry: ConfigEntry
+
+    def show_user_form(
+        self,
+        user_input: dict[str, Any] | None = None,
+        errors: dict[str, str] | None = None,
+        description_placeholders: dict[str, str] | None = None,
+        step_id: str = "user",
+    ) -> ConfigFlowResult:
+        """Show the user form."""
+        if user_input is None:
+            user_input = {}
+        return self.async_show_form(
+            step_id=step_id,
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_URL, default=user_input.get(CONF_URL, "")
+                    ): TextSelector(TextSelectorConfig(type=TextSelectorType.URL)),
+                    vol.Optional(
+                        CONF_MAX_ENTRIES,
+                        default=user_input.get(CONF_MAX_ENTRIES, DEFAULT_MAX_ENTRIES),
+                    ): cv.positive_int,
+                },
+            ),
+            description_placeholders=description_placeholders,
+            errors=errors,
+        )
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a flow initialized by the user."""
+        if not user_input:
+            return self.show_user_form()
+
+        self._async_abort_entries_match({CONF_URL: user_input[CONF_URL]})
+
+        feed = await async_fetch_feed(self.hass, user_input[CONF_URL])
+
+        if feed.bozo:
+            LOGGER.debug("feed bozo_exception: %s", feed.bozo_exception)
+            if isinstance(feed.bozo_exception, urllib.error.URLError):
+                return self.show_user_form(user_input, {"base": "url_error"})
+
+        if not feed.entries:
+            return self.show_user_form(user_input, {"base": "no_feed_entries"})
+
+        feed_title = feed["feed"]["title"]
+
+        return self.async_create_entry(title=feed_title, data=user_input)
+
+    async def async_step_import(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle an import flow."""
+        return await self.async_step_user(user_input)
```

Comment:
We shouldn't return an error for the import flow, instead we should abort and raise a proper corresponding issue for that to comply with ADR-0021